state-sync/storage-service/server/src/lib.rs

```diff
@@ -99,45 +109,99 @@ impl<T: StorageReaderInterface> StorageServiceServer<T> {
             lru_response_cache,
             optimistic_fetches,
             request_moderator,
+            storage_service_listener,
         }
     }
 
+    /// Spawns all continuously running utility tasks
+    async fn spawn_continuous_storage_summary_tasks(&mut self) {
+        // Create a channel to notify the optimistic fetch
+        // handler about updates to the cached storage summary.
+        let (cached_summary_update_notifier, cached_summary_update_listener) =
+            aptos_channel::new(QueueStyle::FIFO, CACHED_SUMMARY_UPDATE_CHANNEL_SIZE, None);
+
+        // Spawn the refresher for the storage summary cache
+        self.spawn_storage_summary_refresher(cached_summary_update_notifier)
```

Comment:
Do we actually need both the storage_summary_refresher and optimistic_fetch_handler? It seems like we want an optimistic fetch if and only if the summary is refreshed, but I may be missing something.

Reply:
> Do we actually need both the storage_summary_refresher and optimistic_fetch_handler?

Can you elaborate? 😄 Is the question more around why we separate them into separate tasks? If so, it's because the optimistic handler can take a non-negligible amount of time to complete, and we don't want it to block the refresher (because this will slow down storage summary updates and affect data advertisements, hence the two tasks). 

> It seems like we want an optimistic fetch if and only if the summary is refreshed, but I may be missing something.

Yeah, this is partly true, but we also need the optimistic fetcher handler to run periodically so that it can perform garbage collection on expired and/or invalid requests. Which is why we still need some periodic execution (e.g., the interval).

Reply:
Makes sense. Thanks!